cmd/util.go

```diff
@@ -304,21 +311,33 @@ func readProject() (*workspace.Project, string, error) {
 	// Now that we got here, we have a path, so we will try to load it.
 	path, err := workspace.DetectProjectPathFrom(pwd)
 	if err != nil {
-		return nil, "", errors.Wrapf(err,
-			"could not locate Pulumi.yaml project file (searching upwards from %s)", pwd)
+		return nil, "", errors.Wrapf(err, "Failed to find current Pulumi project because of "+
+			"an error when searching for the Pulumi.yaml file (searching upwards from %s)", pwd)
 	} else if path == "" {
-		return nil, "", errors.Errorf(
-			"no Pulumi.yaml project file found (searching upwards from %s). If you have not "+
-				"created a project yet, use `pulumi new` to do so", pwd)
+		return nil, "", errReadProjNoPulumiYAML(projType, pwd)
 	}
 	proj, err := workspace.LoadProject(path)
 	if err != nil {
-		return nil, "", err
+		return nil, "", errors.Wrapf(err, "Failed to load Pulumi project located at %q", path)
 	}
 
 	return proj, filepath.Dir(path), nil
 }
 
+func errReadProjNoPulumiYAML(projType projType, pwd string) error {
+	switch projType {
+	case pulumiPolicyProj:
+		return fmt.Errorf("Could not find a Pulumi policy pack project (searching upwards for a "+
```

Comment:
For my own understanding, what is the difference between a Pulumi policy pack project and a Pulumi app? Are they both described via a `Pulumi.yaml` file, and do they have the same schema? If they are indistinguishable, can we just use something generic like "Pulumi project"?

Reply:
We're trying to communicate is that we're looking for the set of code that contains your resource policies, which is different from the set of code that contains your application.

The impetus behind this change is that the error used to be so confusing that it inadvertently misled you into believing you needed to run `pulumi new` and `pulumi stack init` before running `pulumi policy publish`.

I think we're happy to change it to any message that accomplishes the first and avoids the second, but I personally am skeptical that using a more generic, unqualified "pulumi project" makes it less confusing?

Reply:
That's all different from what a Pulumi project _is_, but right now there is not a technical distinction. Someday there will be, but I wanted to keep that issue aside from what we were trying to say in this error message.

Reply:
I don't feel like this is the right change.

The previous message for the normal Pulumi project case was better - we shouldn't change that.

For the policy-pack case - it seems we should just leave off the second sentence.

Even better would be to make `pulumi new` support creating policy packs (just add a template) - then we don't even need to  change the message.

Reply:
Why do you think adding a policy template to `pulumi new` solves this problem? The user still has to find a policy pack template somehow, right? How do they know if they're instantiating the right "kind" of project to begin with? Will there be other project modalities in the future?

Reply:
@lukehoban @chrsmith I've changed the error message back, but now that I think about it, I think this is a mistake. The whole reason we're changing this at all is because this exact message tricked Chris into running `pulumi new`. I think this is simply not going to work for the policy pack case.

Reply:
After more reflection, I don't understand why the other error message I suggested does not do a good job of fixing that specific error. I'd like to use it. If you have a better suggestion we can go with that.

Reply:
I sort of like @lukehoban's feedback here, about keeping the existing message for the project case (which is the 99% case).  For the policy pack case, it seems like just dropping the second sentence would work.

So:

(missing Pulumi.yaml in a project context): "no Pulumi.yaml project file found (searching upwards from %s). If you have not created a project yet, use `pulumi new` to do so"
(missing Pulumi.yaml in a policy pack context): "no Pulumi.yaml project file found (searching upwards from %s)"

FWIW, I really don't like the term "Pulumi app", I don't think we really use this term elsewhere.